client/ayon_core/hosts/houdini/api/hda_utils.py

```diff
@@ -412,3 +412,79 @@ def keep_background_images_linked(node, old_name):
 
     if changes:
         set_background_images(parent, images)
+
+
+def get_products_menu_items(node):
+    """get products menu items
+
+    It gets a list of available products of the specified product types
+      within the specified folder path with in the specified project.
+
+    Users can specify those in the HDA parameters.
+
+    Returns:
+        List[str]: Product options for Products menu.
+    """
+    project_name = node.evalParm("project_name")
+    folder_path = node.evalParm("folder_path")
+    my_product_type = node.evalParm("representation_name")
```

Comment:
Why is `representation_name` the `product_type` according to you? :) Seems like you're missing up terminology?

```suggestion
    representation_name = node.evalParm("representation_name")
```

Reply:
I think it's `product_type`, 
This what I found when I tested it. 
![image](https://github.com/BigRoy/ayon-core/assets/20871534/c3c787de-a7b8-4e56-83e2-fab043c5cbee)

Also, the `product_type` for usd products is the same as their `representation` which is `usd`
![image](https://github.com/BigRoy/ayon-core/assets/20871534/0e273195-621d-4171-8c05-4b30b8d29054)

Reply:
> Also, the product_type for usd products is the same as their representation which is usd

This isn't required to be true. So let's stick to explicitly using the representation name.

Reply:
> This isn't required to be true. So let's stick to explicitly using the representation name.

Ok, I'll change it back in the next commit.

But, My main concern is that some confusion may occur when adopting the same functionality with different products than `usd`

```python
project_name = "some project name"
folder_path = "some folder path"
representation_name = "pointcache"

folder_entity = ayon_api.get_folder_by_path(project_name,
                                                folder_path,
                                                fields={"id"})

products = ayon_api.get_products(
        project_name,
        folder_ids=[folder_entity["id"]],
        product_types=[representation_name]
    )
```